torch_geometric/datasets/lrgb.py

```diff
@@ -0,0 +1,293 @@
+        path = osp.join(self.processed_dir, f'{split}.pt')
+        self.data, self.slices = torch.load(path)
+
+    @property
+    def raw_dir(self) -> str:
+        return osp.join(self.root, self.name, 'raw')
+
+    @property
+    def processed_dir(self) -> str:
+        return osp.join(self.root, self.name, 'processed')
+
+    @property
+    def raw_file_names(self) -> List[str]:
+        if self.name.split('-')[1] == 'sp':
+            return ['train.pickle', 'val.pickle', 'test.pickle']
+        else:
+            return ['train.pt', 'val.pt', 'test.pt']
+
+    @property
+    def processed_file_names(self) -> List[str]:
+        return ['train.pt', 'val.pt', 'test.pt']
+
+    def download(self):
+        shutil.rmtree(self.raw_dir)
+        path = download_url(self.urls[self.name], self.root)
+        extract_zip(path, self.root)
+        os.rename(osp.join(self.root, self.dwnld_file_name[self.name]),
+                  self.raw_dir)
+        os.unlink(path)
+
+    def process(self):
+        if self.name == 'pcqm-contact':
+            # PCQM-Contact
+            self.process_pcqm_contact()
+        else:
+            # For COCO-SP to remap the labels as the
+            # original label idxs are not contiguous
+            label_map_coco = LabelEncoder().fit(self.original_label_idxs())
+            for split in ['train', 'val', 'test']:
+                if self.name.split('-')[1] == 'sp':
+                    # PascalVOC-SP and COCO-SP
+                    with open(osp.join(self.raw_dir, f'{split}.pickle'),
+                              'rb') as f:
+                        graphs = pickle.load(f)
+                elif self.name.split('-')[0] == 'peptides':
+                    # Peptides-func and Peptides-struct
+                    with open(osp.join(self.raw_dir, f'{split}.pt'),
+                              'rb') as f:
+                        graphs = torch.load(f)
+
+                indices = range(len(graphs))
+
+                pbar = tqdm(total=len(indices))
+                pbar.set_description(f'Processing {split} dataset')
+
+                data_list = []
+                for idx in indices:
+                    graph = graphs[idx]
+
+                    if self.name.split('-')[1] == 'sp':
+                        """
+                        PascalVOC-SP and COCO-SP
+                        Each `graph` is a tuple (x, edge_attr, edge_index, y)
+                            Shape of x : [num_nodes, 14]
+                            Shape of edge_attr : [num_edges, 2]
+                            Shape of edge_index : [2, num_edges]
+                            Shape of y : [num_nodes]
+                        """
+                        x = graph[0].to(torch.float)
+                        edge_attr = graph[1].to(torch.float)
+                        edge_index = graph[2]
+                        if self.name == 'coco-sp':
+                            y = label_map_coco.transform(graph[3])
```

Comment:
I think we can directly use `LabelEncoder.fit_transform(graph[3])` here without the need of `original_label_idxs`, wdyt?

Reply:
This is because one graph sample will only have a subset of the labels present in `original_label_idxs`, and hence would require to fit on the full label set.

Reply:
Oh, I see. You are right!